src/run_environment/machine.ml

```diff
@@ -122,6 +122,11 @@ module Make_fun = struct
                                   ["#WIP"]). *)
       | `Custom of string (** Pass arbitrary data (useful for temporary
                               extensions/experiements outside of Biokepi). *)
+      | `Self_identification of string list
```

Comment:
It would be very annoying to manage:

- the `?requirements` (the argument needs to go up the stack very far, you would end-up with `Gatk.indel_realigner` that takes tons of overriding arguments: `?gatk_requirements`, `?samtools_sort_requirements`, `?samtools_index_requirements`, `?merge_bams_requirements`, etc. (installations, downloads, …)
- and then it would have to go up the `Pipeline_edsl` API too

And on top of that it would break the clean-ish separation between bioinformatics and infrastructure. The goal is to be able to reuse 100% of the pipeline/`Tools` code unchanged, and switch between completely different `Machine.t`s at will.